File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_set<int> s;
        for(const int& e : nums) {
            if (s.find(e) != s.end()) return true;
            s.insert(e);
        }
        return false;
    }
};

File: Data Structures & Algorithms/is-subsequence/submission-0.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        i, j = 0, 0
        n, m = len(t), len(s)
        while i < n and j < m:
            if s[j] == t[i]:
                j += 1
            i += 1
        return j == m